app/src/main/java/com/findmyrecycling/MainActivity.kt

```diff
@@ -75,38 +78,48 @@ fun RecycleSearch(product: String) {
 @Preview(showBackground = true)
 @Composable
 fun DefaultPreview() {
-    FindMyRecyclingTheme {
+    Column (Modifier.fillMaxSize().padding(20.dp), horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.spacedBy(20.dp)) {
         RecycleSearch("Android")
-    }
+        Button(
 
-    Button(
-        onClick = {
-            signIn()
+            onClick = {
+                signIn()
+            }
+        ) {
+            Text(text = "Logon")
         }
-    ) {
-        Text(text = "Logon")
     }
 }
 
+interface FirebaseAuthManager {
+
+    fun buildLoginIntent(): Intent
+    fun buildLoginActivityResult(): FirebaseAuthUIActivityResultContract =
+        FirebaseAuthUIActivityResultContract()
+
+    fun onLoginResult(result: FirebaseAuthUIAuthenticationResult)
+}
+
 fun signIn() {
     val providers = arrayListOf(AuthUI.IdpConfig.EmailBuilder().build())
     val signinIntent = AuthUI.getInstance()
         .createSignInIntentBuilder()
         .setAvailableProviders(providers)
         .build()
 
-    signInLauncher.launch(signinIntent)
+//    signInLauncher.launch(signinIntent)
```

Comment:
Avoid pushing commented code to Version Control.  Version control has tools that makes this unnecessary.
- You can look at history to see what used to be there.
- You can use branches for experimental features.